### PR TITLE
Example Lens

### PR DESCRIPTION
**Pull request by jgraff2**

### CODE REVIEW

File: Lenses/Example/src/lens.css [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+.subject .column {
+  flex: 1;
+}
+
+.subject .column.samples {
+  border-left: thin solid;
+}
+
+/* override lightning css */
```
Comment (jgraff2) [inline Lenses/Example/src/lens.css]: The lightning css interferes with the lens css when loaded into a perspective. Is there a better way to do this? Are there plans to restructure things so this isn't necessary?

File: Lenses/Example/src/lens.css [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+ul {
+  margin: 0;
+  padding-left: 1em;
+  list-style-type: disc;
+}
+
```
Comment (jgraff2) [inline Lenses/Example/src/lens.css]: The lightning css interferes with the lens css when loaded into a perspective. Is there a better way to do this? Are there plans to restructure things so this isn't necessary?

File: Lenses/Example/src/lens.css [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  color: inherit;
+  text-decoration: none;
+}
+
+.top-link:hover {
+  font-weight: bold;
+  text-decoration: none;
+}
+
+.OK {
```
Comment (iamigo) [inline Lenses/Example/src/lens.css]: I prefer something like this (it leaves less room for possible conflicts):

```
#lens .status-OK {
  color: green;
}
```

etc.

**Comment by iamigo** [comment]
@jgraff2 I updated main.js with a bunch of comments--does that look ok?

File: Lenses/Example/src/lens.css [hunk 4]
```diff
@@ -2,6 +2,7 @@
  * lens.css
  *
  * The stylesheet for the Example lens.
+ * All rules should be prefixed by #lens
```
Comment (jgraff2) [inline Lenses/Example/src/lens.css]: I added a comment here - anything to add?

File: Lenses/Example/src/main.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+require('./lens.css');
+
+/*
+ * We use handlebars for templates--you can read more about handlebars at
+ * http://handlebarsjs.com/.
+ */
+const handlebars = require('handlebars-template-loader/runtime');
+const loadingTemplate = require('./template/loading.handlebars');
+const mainTemplate = require('./template/main.handlebars');
+handlebars.registerPartial('subject',
```
Comment (jgraff2) [inline Lenses/Example/src/main.js]: Did you mean to change this one and not the other two?

File: Lenses/Example/src/main.js [hunk 6]
```diff
@@ -1,47 +1,103 @@
...
 
+/*
+ * We use handlebars for templates--you can read more about handlebars at
+ * http://handlebarsjs.com/.
+ */
 const handlebars = require('handlebars-template-loader/runtime');
 const loadingTemplate = require('./template/loading.handlebars');
 const mainTemplate = require('./template/main.handlebars');
-handlebars.registerPartial('subject', require('./template/subject.handlebars'));
+handlebars.registerPartial('subject',
```
Comment (jgraff2) [inline Lenses/Example/src/main.js]: Did you mean to change this one and not the other two?

File: Lenses/Example/src/main.js [hunk 7]
```diff
@@ -1,47 +1,103 @@
...
+  require('./template/subject.handlebars'));
 handlebars.registerPartial('aspect', require('./template/aspect.handlebars'));
 handlebars.registerPartial('sample', require('./template/sample.handlebars'));
 
 handlebars.registerHelper('listClass', (array) => {
-  if (array == null) return '';
-  if (array.length <= 3) return 'inline';
-  else return 'multiline';
+  if (array == null) {
```
Comment (jgraff2) [inline Lenses/Example/src/main.js]: Is this another code style rule I should remember? No single-line ifs?

File: Lenses/Example/src/main.js [hunk 8]
```diff
@@ -1,47 +1,103 @@
...
+/*
+ * Once the lens is loaded, we register event listeners for some of the other
+ * refocus events.
+ */
 LENS.addEventListener('refocus.lens.load', () => {
   LENS.addEventListener('refocus.lens.hierarchyLoad', onHierarchyLoad);
   LENS.addEventListener('refocus.lens.realtime.change', onRealtimeChange);
+
+  /*
+   * Display a "Loading..." indicator while we wait for the browser to get all
```
Comment (jgraff2) [inline Lenses/Example/src/main.js]: Currently the text is "LOADING". Should I change it to "Loading..."?

**Comment by jgraff2** [comment]
@iamigo did you see my 3 comments on main.js?

**Comment by iamigo** [comment]
No, @jgraff2, I didn't see any comments on main.js here--sorry :( was I too hasty in merging?

File: Lenses/Example/src/main.js (hunk 6 above)
Comment (jgraff2) [inline Lenses/Example/src/main.js]: @iamigo

File: Lenses/Example/src/main.js (hunk 7 above)
Comment (jgraff2) [inline Lenses/Example/src/main.js]: @iamigo

File: Lenses/Example/src/main.js (hunk 8 above)
Comment (jgraff2) [inline Lenses/Example/src/main.js]: @iamigo

File: Lenses/Example/src/main.js (hunk 6 above)
Comment (iamigo) [inline Lenses/Example/src/main.js]: I only changed the one which violated the "max 80 chars per line" rule

File: Lenses/Example/src/main.js (hunk 7 above)
Comment (iamigo) [inline Lenses/Example/src/main.js]: Correct.

File: Lenses/Example/src/main.js (hunk 8 above)
Comment (iamigo) [inline Lenses/Example/src/main.js]: I don't feel too strongly about it either way

File: Lenses/Example/src/main.js (hunk 6 above)
Comment (jgraff2) [inline Lenses/Example/src/main.js]: ah ok

**Pull request merged**